### PR TITLE
PowerShell Artifact Examples

### PR DESCRIPTION
**Pull request by coryreid**
* Moved `UploadArtifact.ps1` to new Artifacts folder
* Removed extraneous `$spaceName` variable from `UploadArtifact.ps1`
* Added `DownloadArtifact.ps1` script example

### CODE REVIEW

File: REST/PowerShell/Artifacts/DownloadArtifact.ps1 [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+Write-Host "Getting the release information"
+$releaseList = Invoke-RestMethod -Method Get -Uri "$OctopusUrl/api/$spaceId/projects/$projectId/releases?skip=0&take=100&searchByVersion=$releaseVersion" -Headers $header
+$release = $releaseList.Items | Where-Object {$_.Version -eq $releaseVersion}
+$releaseId = $release.Id
+Write-Host "The id of $releaseVersion is $releaseId"
+
+Write-Host "Getting the deployment information"
+$deploymentList = Invoke-RestMethod -Method Get -Uri "$OctopusUrl/api/$spaceId/releases/$releaseId/deployments?skip=0&take=1000" -Headers $header
+$deploymentsToEnivronment = @($deploymentList.Items | Where-Object {$_.EnvironmentId -eq $environmentId})
```
Comment (harrisonmeister) [inline REST/PowerShell/Artifacts/DownloadArtifact.ps1:38]:
```suggestion
$deploymentsToEnvironment = @($deploymentList.Items | Where-Object {$_.EnvironmentId -eq $environmentId})
```

File: REST/PowerShell/Artifacts/DownloadArtifact.ps1 [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+$deploymentToUse = $null
+$previousDate = Get-Date
+$previousDate = $previousDate.AddDays(-10000)
+
+foreach ($deployment in $deploymentsToEnivronment)
```
Comment (harrisonmeister) [inline REST/PowerShell/Artifacts/DownloadArtifact.ps1:43]:
```suggestion
foreach ($deployment in $deploymentsToEnvironment)
```

File: REST/PowerShell/Artifacts/DownloadArtifact.ps1 [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+Write-Host "The id of $releaseVersion is $releaseId"
+
+Write-Host "Getting the deployment information"
+$deploymentList = Invoke-RestMethod -Method Get -Uri "$OctopusUrl/api/$spaceId/releases/$releaseId/deployments?skip=0&take=1000" -Headers $header
+$deploymentsToEnivronment = @($deploymentList.Items | Where-Object {$_.EnvironmentId -eq $environmentId})
+$deploymentToUse = $null
+$previousDate = Get-Date
+$previousDate = $previousDate.AddDays(-10000)
+
+foreach ($deployment in $deploymentsToEnvironment)
```
Comment (harrisonmeister) [inline REST/PowerShell/Artifacts/DownloadArtifact.ps1:43]: There's probably a one-liner way of doing this (getting the earliest). Something like:

```powershell
$deploymentToUse = $deploymentsToEnvironment | Sort-Object {[DateTime]$_."Created"} | Select -First 1
```

**Review by harrisonmeister — COMMENTED** [review]
Made one suggestion. Apart from that LGTM

File: REST/PowerShell/Artifacts/DownloadArtifact.ps1 (hunk 3 above)
Comment (coryreid) [reply REST/PowerShell/Artifacts/DownloadArtifact.ps1]: Agreed, added this to both 👍

**Review by harrisonmeister — APPROVED** [review]
:+1:

**Pull request merged**